test/multiarch_utils.go

```diff
@@ -0,0 +1,131 @@
+		mapping[Registry] = "registry"
+	}
+	return mapping
+}
+
+func initExcludedTests() map[string]bool {
+	mapping := make(map[string]bool)
+	tests := []string{}
+	switch arch := runtime.GOARCH; arch {
+	case "s390x":
+		//examples
+		tests = []string{
+			"v1alpha1/taskruns/dind-sidecar",
+			"v1beta1/taskruns/dind-sidecar",
+			"v1alpha1/taskruns/build-gcs-targz",
+			"v1beta1/taskruns/build-gcs-targz",
+			"v1alpha1/taskruns/build-push-kaniko",
+			"v1alpha1/taskruns/pull-private-image",
+			"v1beta1/taskruns/pull-private-image",
+			"v1alpha1/pipelineruns/pipelinerun",
+			"v1beta1/pipelineruns/pipelinerun",
+			"v1beta1/taskruns/build-gcs-zip",
+			"v1alpha1/taskruns/build-gcs-zip",
+			"v1alpha1/taskruns/git-volume",
+			"v1beta1/taskruns/git-volume",
+			"v1beta1/taskruns/docker-creds",
+			"v1alpha1/taskruns/docker-creds",
+			"v1beta1/taskruns/steps-run-in-order",
+			"v1alpha1/taskruns/steps-run-in-order",
+			"v1beta1/taskruns/step-by-digest",
+			"v1alpha1/taskruns/step-by-digest",
+			"v1alpha1/taskruns/gcs-resource",
+			"v1beta1/taskruns/gcs-resource",
+			"v1beta1/taskruns/authenticating-git-commands",
+			"v1beta1/pipelineruns/pipelinerun-with-final-tasks",
+			"v1beta1/taskruns/pullrequest_input_copystep_output",
+			"v1alpha1/taskruns/pullrequest_input_copystep_output",
+			"v1beta1/taskruns/pullrequest",
+			"v1alpha1/taskruns/pullrequest",
+			"v1beta1/taskruns/step-script",
+			"v1alpha1/taskruns/step-script",
+			"v1beta1/pipelineruns/conditional-pipelinerun",
+			"v1alpha1/pipelineruns/pipelinerun-with-resourcespec",
+			"v1beta1/pipelineruns/pipelinerun-with-resourcespec",
+			"v1beta1/taskruns/git-ssh-creds-without-known_hosts",
+			"v1alpha1/taskruns/optional-resources",
+			"v1beta1/taskruns/optional-resources",
+			"v1beta1/taskruns/task-output-image",
+			//e2e
+			"TestEntrypointRunningStepsInOrder",
+			"TestWorkingDirIgnoredNonSlashWorkspace",
+			"TestTaskRun_EmbeddedResource",
+			"TestTaskRunPipelineRunCancel",
+			"TestEntrypointRunningStepsInOrder",
+			"TestGitPipelineRun",
+			"TestGitPipelineRunFail",
+			"TestGitPipelineRunWithRefspec",
+			"TestGitPipelineRun_Disable_SSLVerify",
+			"TestGitPipelineRunWithNonMasterBranch",
+			"TestHelmDeployPipelineRun",
+			"TestKanikoTaskRun",
+			"TestPipelineRun",
+			"TestSidecarTaskSupport",
+			"TestWorstkingDirCreated",
+			"TestWorkingDirIgnoredNonSlashWorkspace",
+			"TestWorkingDirCreated",
+			"service account propagation and pipeline param",
+			"pipelinerun succeeds with LimitRange minimum in namespace",
+		}
+	default:
+		// do nothing
+	}
+
+	for _, test := range tests {
+		mapping[test] = true
+	}
+
+	return mapping
+}
+
+// get test image based on unique id
+func GetTestImage(image int) string {
+	return imageNames[image]
+}
+
+// check if test name is in the excluded list based on current hardware architecture
+func IsExcluded(testName string) bool {
+	return excludedTests[testName]
```

Comment:
Do we ever expect this to be false? Would it be easier to instead make this a map[string]struct{} and use an existence check?

Reply:
yes, we expect this to be false for some cases. Following https://blog.golang.org/maps `map[string]bool` returns false if key doesn't exist in the map.
You also can have a look in the integration test, no tests from the "arch specific list" were skipped

Reply:
Ah, thanks for clarifying.